TE-1362 Extend `Reviews` with submit form

red-runner opened the pull request:
[Related YouTrack issue](https://youtrack.lodgify.net/issue/TE-1362)

### What **one** thing does this PR do?
* Implements a submit form behaviour for the `Reviews` component; described in this [YouTrack issue](https://youtrack.lodgify.net/issue/TE-1339)

### Any other notes
* It also upgrades the `RatingInput` component exposing a label prop

#### Default

![standard reviews](https://user-images.githubusercontent.com/33876435/48778377-500cb780-ecd5-11e8-9b79-7e0d150148e3.gif)

#### Content

![content_strings](https://user-images.githubusercontent.com/33876435/48778405-631f8780-ecd5-11e8-86ae-860f7fb1b37b.gif)

#### Usage Validation

![usage_validation](https://user-images.githubusercontent.com/33876435/48778446-792d4800-ecd5-11e8-8357-a25f1f0d32b2.gif)

#### State Success

![state_success](https://user-images.githubusercontent.com/33876435/48778490-98c47080-ecd5-11e8-931b-363b1b27705f.gif)

#### State Error

![state_error](https://user-images.githubusercontent.com/33876435/48778513-ac6fd700-ecd5-11e8-8389-3ca923e55cda.gif)

marcdavi-es commented on src/components/property-page-widgets/Reviews/Readme.md:
```diff
@@ -2,21 +2,35 @@
-      source: 'The Owner',
-      text:
-        'You can also personally respond to each review: Thanks for you kind review, James! Hope to welcome you back soon!',
-    },
```
Retain the `reviewResponse` for this first basic example, but exclude it from the others to reduce noise.

marcdavi-es commented on src/components/inputs/RatingInput/component.js:
```diff
@@ -1,35 +1,41 @@
   error: false,
   iconSize: 'small',
-  name: '',
+  label: RATINGINPUT_TEXT,
```
I think in other inputs, default `label` value is `''` ?

marcdavi-es commented on src/components/inputs/RatingInput/component.js:
```diff
@@ -38,6 +44,7 @@ Component.propTypes = {
   error: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
   /** The size of the star icons. */
   iconSize: PropTypes.oneOf(['tiny', 'small']),
+  label: PropTypes.string,
```
Needs a prop doc comment.

marcdavi-es commented on src/components/inputs/RatingInput/constants.js:
```diff
@@ -1,2 +1,3 @@
 export const MAX_RATING = 5;
 export const ONRATE_HANDLER = 'onRate';
```
and `RATING_INPUT_TEXT`

marcdavi-es commented on src/utils/default-strings/constants.js:
```diff
@@ -37,6 +40,7 @@ export const PHONE = 'Phone';
 export const PRICE_PER_EXTRA_PER = 'Price per extra per.';
 export const PROPERTY = 'Property';
 export const PROPERTY_PICTURES = 'Property pictures';
+export const REVIEW_FORM_HEADING = 'Add a Review';
```
All of these identifiers match the strings. This should be 

```js
export const ADD_A_REVIEW = 'Add a Review';
```

red-runner commented:
@marcdavi-es  comments are solved

stephenkr commented on src/components/property-page-widgets/Reviews/utils/getModalFormMarkup.js:
```diff
@@ -0,0 +1,101 @@
+ * @param {string} props.emailInputLabel
+ * @param {string} props.errorMessage
+ * @param {string} props.guestTypeInputLabel
+ * @param {Object[]} props.guestTypeOptions
```
Can you vertically align these props?

stephenkr commented on src/components/property-page-widgets/Reviews/utils/getModalFormMarkup.js:
```diff
@@ -0,0 +1,101 @@
+  titleInputLabel,
+  validation,
+  yearInputLabel,
+  yearsOption,
```
Don't forget to add in the eslint enable comment after the props.
```
/* eslint-enable react/prop-types */
```

stephenkr commented on src/components/property-page-widgets/Reviews/utils/getModalFormMarkup.spec.js:
```diff
@@ -0,0 +1,50 @@
+  yearOptions,
+};
+
+const getModalForm = props => mount(<div>{getModalFormMarkup(props)}</div>);
```
I don't think you need to wrap the utility with the `div` tags, I think `mount(getModalFormMarkup(props))` is enough.

marcdavi-es commented on src/components/property-page-widgets/Reviews/component.js:
```diff
@@ -65,17 +78,115 @@ export const Component = ({
+    })
+  ),
   /** The text to display as a heading at the top of the widget. */
+  // eslint-disable-next-line react/no-unused-prop-types
```
No need for this.

marcdavi-es commented on src/components/property-page-widgets/Reviews/component.js:
```diff
@@ -65,17 +78,115 @@ export const Component = ({
+  // eslint-disable-next-line react/no-unused-prop-types
+  reviewFormHeading: PropTypes.string,
   /** The collection of reviews. */
+  // eslint-disable-next-line react/no-unused-prop-types
```
Same as https://github.com/lodgify/lodgify-ui/pull/439/commits/905f33e464d0836a757fdd3b1546e85585924718#r235345386

marcdavi-es commented on src/components/property-page-widgets/Reviews/component.js:
```diff
@@ -65,17 +78,115 @@ export const Component = ({
   ratingAverage: PropTypes.number.isRequired,
+  /** A visible label to display with the rating stars. */
+  // eslint-disable-next-line react/no-unused-prop-types
+  ratingText: PropTypes.string,
```
This should be called `ratingInputLabel`, right?

marcdavi-es commented on src/components/property-page-widgets/Reviews/component.js:
```diff
@@ -65,17 +78,115 @@ export const Component = ({
+  ratingText: PropTypes.string,
+  /** The text to display as a heading at the top of the modal form */
+  // eslint-disable-next-line react/no-unused-prop-types
+  reviewFormHeading: PropTypes.string,
```
`reviewFormHeadingText`

marcdavi-es commented on src/components/property-page-widgets/Reviews/component.js:
```diff
@@ -104,5 +215,42 @@ Component.propTypes = {
+  yearInputLabel: PropTypes.string,
+  /** The options which the user can select for the year field. */
+  // eslint-disable-next-line react/no-unused-prop-types
+  yearsOption: PropTypes.arrayOf(
```
`yearsOptions`

marcdavi-es commented on src/utils/default-strings/constants.js:
```diff
@@ -37,6 +40,7 @@ export const PHONE = 'Phone';
 export const PRICE_PER_EXTRA_PER = 'Price per extra per.';
 export const PROPERTY = 'Property';
 export const PROPERTY_PICTURES = 'Property pictures';
+export const ADD_A_REVIEW = 'Add a Review';
```
Dude - this whole file is alphabetical for ease of indexing.

stephenkr commented on package.json:
```diff
@@ -20,7 +20,7 @@
   "size-limit": [
     {
-      "limit": "258 KB",
+      "limit": "265 KB",
```
Can you set this to `259 KB`?

Looking at the logs in the CI, you can see the returned bundle size is 258.73 KB.

<img width="484" alt="screen shot 2018-11-22 at 10 43 56" src="https://user-images.githubusercontent.com/25742275/48894715-b1eb2f80-ee43-11e8-8c4d-9710838896bd.png">

marcdavi-es replied:
@stephenkr Why?

stephenkr replied:
I thought we were trying to keep the bundle size limit as small as possible?

marcdavi-es replied:
But we nudge it up every time we exceed the 'limit'.

Right now we don't have a budget in force in practice - so there's not substantive difference between increasing this by 1KB or 6KB.

marcdavi-es commented on src/components/inputs/RatingInput/component.js:
```diff
@@ -1,35 +1,41 @@
   iconSize: 'small',
-  name: '',
+  label: '',
+  name: 'ratingInput',
```
This isn't consistent with any of the other inputs.

red-runner commented:
@marcdavi-es 
* Changed `Readme` for the `RatingInput` to make it more consistent with other inputs.
* Set name `defaultProp` for the `RatingInput` to `''` to make it more consistent with other inputs.
* Changed `Readme` for the `Reviews` to make it more consistent with other components with forms.
* I've made validation object explicit in the `Reviews` `Readme` to make it more explicit. 
* Added `name="rating"` in the `getModalFormMarkup.js` to make it consistent with the rest of the inputs. 
* Solved indentation issues in `Readme` files, wrong spacing etc.
